### ISSUE TITLE
[Small Bug] The embedding fusion is not necessary for LLaMA models. 

### ISSUE THREAD

**Issue by ChenMnZ** [spcl/QuaRot#7]
`LlamaRMSNorm` do not contain meaning operation. Therefore, the LLaMA model family do not require the fusion of embedding as follows.

https://github.com/spcl/QuaRot/blob/d538b8ef8127fe199d3fc5e43bbfa696e637ba92/fake_quant/rotation_utils.py#L51-L54

This is just a minor error and almost no impact on the outcome.

**Comment by sashkboos** [comment]
Thanks @ChenMnZ 

This is of course is not important in LLaMa models. However, I will keep it as we plan to support other models (like OPT) as well. 

If this addresses your issue, I will close it :)

**Comment by mxjmtxrm** [comment]
Hi,
I'm confused as to why embedding fusion is needed, and why it does not affect llama performance?

**Comment by L1aoXingyu** [comment]
> Hi, I'm confused as to why embedding fusion is needed, and why it does not affect llama performance?

because llama embedding mean(dim=1) almost are very small numbers.

```
>> print(model.model.model.embed_tokens.weight.data.mean(dim=1)[:10])
>> tensor([ 2.9802e-07, -4.1723e-07,  0.0000e+00,  2.3842e-07, -1.7881e-07,
         7.7486e-07,  8.3447e-07,  4.1723e-07, -0.0000e+00, -2.3842e-07],
       dtype=torch.float16)
```

**Comment by wenhuach21** [comment]
Could you kindly elaborate on why it's necessary to subtract the mean from the embedding and the following code for opt model?
~~~python
  if model_type == model_utils.OPT_MODEL:
            bake_mean_into_linear(layer.self_attn.out_proj)
            bake_mean_into_linear(layer.fc2)
                    
~~~

**Comment by sashkboos** [comment]
@wenhuach21 

This is because the normal layer norm (which is used in OPT) has mean subtraction as it's first step (see https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html), but [LLaMa-RMSNorm](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L111) does not.

**Comment by wenhuach21** [comment]
> @wenhuach21
> 
> This is because the normal layer norm (which is used in OPT) has mean subtraction as it's first step (see https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html), but [LLaMa-RMSNorm](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L111) does not.

Got it. Thank you very much!
